sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanPartitioning.scala

```diff
@@ -32,15 +32,15 @@ import org.apache.spark.util.collection.Utils.sequenceToOption
  */
 object V2ScanPartitioning extends Rule[LogicalPlan] with SQLConfHelper {
   override def apply(plan: LogicalPlan): LogicalPlan = plan transformDown {
-    case d @ DataSourceV2ScanRelation(relation, scan: SupportsReportPartitioning, _, _) =>
+    case d @ DataSourceV2ScanRelation(relation, scan: SupportsReportPartitioning, _, None) =>
```

Comment:
Do you think it's sufficient to only match keyGroupedPartitioning = None?

Reply:
After second thought, I think it should only match `None` here, otherwise `catalystPartitioning` will be calculated every round, if the generated `catalystPartitioning` contains `Alias`, will cause `!plan.fastEquals(reOptimized)` and fail `checkBatchIdempotence`.

"SPARK-30289 Create: partitioned by nested column" happen to cover this case after we changed the `InMemoryTable#outputPartitioning`

Reply:
Yes, I think it's better to match `None` here.